### PR TITLE
Apply fixes to `RealmLive<T>` to more correctly support querying data within `PerformRead`

### PR DESCRIPTION
**Pull request by peppy**
This applies what I have learned in https://github.com/realm/realm-dotnet/discussions/2734#discussioncomment-1705038.

### CODE REVIEW

File: osu.Game/Database/RealmLive.cs [hunk 1]
```diff
@@ -102,10 +102,11 @@ public T Value
...
+                if (!isCorrectThread)
+                    throw new InvalidOperationException($"Can't use {nameof(Value)} unless on the same thread the original data was fetched from.");
 
-                using (var realm = Realm.GetInstance(data.Realm.Config))
-                    retrieved = realm.Find<T>(ID);
+                var realm = Realm.GetInstance(data.Realm.Config);
+                var retrieved = realm.Find<T>(ID);
 
                 if (!retrieved.IsValid)
                     throw new InvalidOperationException("Attempted to access value without an open context");
```
Comment (smoogipoo) [inline osu.Game/Database/RealmLive.cs:112]: This check will never fail now. The whole point is that it was guaranteed to occur after the using block disposes of the Realm.

**Review by smoogipoo — CHANGES_REQUESTED** [review]
Not sure how I feel about this yet...

File: osu.Game/Database/RealmLive.cs (hunk 1 above)
Comment (peppy) [reply osu.Game/Database/RealmLive.cs:112]: Yep, correct. Can be commented for now if preferred.

Comment (bdach) [reply osu.Game/Database/RealmLive.cs:112]: Definitely agree with leaving some sort of info that the exception branch is unreachable.

File: osu.Game.Tests/Database/RealmLiveTests.cs [hunk 2]
```diff
@@ -66,19 +66,29 @@ public void TestValueAccessWithOpenContext()
 
                 Task.Factory.StartNew(() =>
                 {
-                    Assert.DoesNotThrow(() =>
+                    // TODO: The commented code is the behaviour we hope to obtain, but is temporarily disabled.
+                    // See https://github.com/ppy/osu/pull/15851
```
Comment (bdach) [inline osu.Game.Tests/Database/RealmLiveTests.cs:70]: I'm not super sure about these todos. Given the realm team's answer, is the "desirable" behaviour ever actually going to be within reach? If not, then at that point the code could as well be removed as it serves little purpose.

Alternatively these paths could be shipped off to `[Ignore]`d tests. But I'm still not sure they would give much value.

Comment (peppy) [reply osu.Game.Tests/Database/RealmLiveTests.cs:70]: I'll remove it all completely for now.

**Comment by peppy** [comment]
The real question is: do we even leave `ILive.Value` as existing? When I was making this PR I considered removing it, but if you look through the usages, it will required 50+ changes to cases which are mostly in tests, and code readability will take a hit. Not sure.

**Comment by bdach** [comment]
I would personally agree with keeping `Value` as a shorthand notation at least. Having isolated find calls everywhere feels like a step back in code quality.

**Comment by peppy** [comment]
I restructured a bit since making that comment. `.Value` still has.. value on the update thread.

Also the proposal would have been replacing it with `.PerformRead(...)` calls. But I think it's fine to leave as-is since things still pass and what not.

**Review by smoogipoo — APPROVED** [review]
Lgtm

**Pull request merged**